File: 7-kyu/jumping-number-(special-numbers-series-#4)/c++/solution.cpp
#include <cmath>
#include <string>

std::string jumpingNumber(int number)
{
    int prev = number % 10;
    number /= 10;
    while (number > 0) {
        int curr = number % 10;
        if (std::abs(curr - prev) != 1) return "Not!!";
        prev = curr;
        number /= 10;
    }
    return "Jumping!!";
}

File: 7-kyu/jumping-number-(special-numbers-series-#4)/javascript/solution.js
const jumpingNumber = n => {
  if (n < 10) {
    return "Jumping!!";
  }
  const nStr = n.toString();
  for (let i = 0; i < nStr.length - 1; i++) {
    if (Math.abs(Number(nStr[i]) - Number(nStr[i + 1])) !== 1) {
      return "Not!!";
    }
  }
  return "Jumping!!";
};
